src/main/java/com/squareup/kotlinpoet/Util.kt

```diff
@@ -15,54 +15,35 @@
  */
 package com.squareup.kotlinpoet
 
-import java.lang.Character.isISOControl
-import java.util.ArrayList
-import java.util.Arrays
-import java.util.Collections
-import java.util.LinkedHashMap
-import java.util.LinkedHashSet
-
 internal object NullAppendable : Appendable {
   override fun append(charSequence: CharSequence) = this
   override fun append(charSequence: CharSequence, start: Int, end: Int) = this
   override fun append(c: Char) = this
 }
 
-internal fun <K, V> Map<K, List<V>>.toImmutableMultimap(): Map<K, List<V>> {
-  val result = LinkedHashMap<K, List<V>>()
-  for ((key, value) in this) {
-    if (value.isEmpty()) continue
-    result.put(key, value.toImmutableList())
-  }
-  return Collections.unmodifiableMap(result)
-}
-
-internal fun <K, V> Map<K, V>.toImmutableMap(): Map<K, V>
-    = Collections.unmodifiableMap(LinkedHashMap(this))
+internal fun <K, V> Map<K, V>.toImmutableMap(): Map<K, V> = LinkedHashMap(this)
```

Comment:
That's clever! Since there are no mutators on `kotlin.collections.Map`, no need to bother with `unmodifiableMap()`

Reply:
It was for Java users

Reply:
Oh yeah, right

Reply:
But that's `internal` anyway?

Reply:
This is used for exposed types

Reply:
Oh, right. Will revert, thanks!

Reply:
Is this a valid use case for a multiplatform implementation?

- `kotlinpoet-common`

```kotlin
expect internal fun <K, V> Map<K, V>.toImmutableMap(): Map<K, V>
```

- `kotlinpoet`

```kotlin
actual internal fun <K, V> Map<K, V>.toImmutableMap(): Map<K, V> = 
    LinkedHashMap(this)
```

- `kotlinpoet-jvm`

```kotlin
actual internal fun <K, V> Map<K, V>.toImmutableMap(): Map<K, V> = 
    Collections.unmodifiableMap(LinkedHashMap(this))
```

Reply:
Yes. But the multiplatform stuff is probably not stable enough to bother. When it's more mature we can.

Reply:
Kotlin and Java are both JRE targets. It'd be weird to have separate artifacts for them.